### PR TITLE
[Docs] [Serve] Has a consistent landing page style

### LINKED ISSUE #25127: [Docs] [Serve] Have a consistent landing page style
### What happened + What you expected to happen

Serve landing page: https://docs.ray.io/en/master/serve/index.html

Datasets landing page: https://docs.ray.io/en/master/data/dataset.html

I think Serve should also have different "blocks" for *Getting Started* etc just like Datasets

### Versions / Dependencies

<= 1.13.0

### Reproduction script

None (this is a Docs issue)

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

### PR DESCRIPTION
**Pull request by emjames**
## Why are these changes needed?
Serve and Dataset landing pages are inconsistent. This change adds `panels` in the [Serve's](https://docs.ray.io/en/master/serve/index.html) "Learn More" section, just like [Dataset's](https://docs.ray.io/en/master/data/dataset.html). This documentation makes it easier for the user to learn more about Serve with Key Concepts, User Guide, Examples, and more.

## Related issue number
Closes #25127

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - N/A: This is a documentation change

### CODE REVIEW

File: doc/source/serve/index.md
Comment (edoakes) [inline doc/source/serve/index.md:163]:
```suggestion
    **User Guides**
```

File: doc/source/serve/index.md
Comment (edoakes) [inline doc/source/serve/index.md:140]:
```suggestion
    Start with our quick start tutorials for :ref:`deploying a single model locally<getting-started>` and how to :ref:`convert an existing model into a Ray Serve deployment<converting-to-ray-serve-deployment>` .
```

File: doc/source/serve/index.md
Comment (edoakes) [inline doc/source/serve/index.md:154]:
```suggestion
    Learn about :ref:`Deployments<serve-key-concepts-deployment>`, :ref:`how to query them<serve-key-concepts-query-deployment>`, and the :ref:`Deployment Graph<serve-key-concepts-deployment-graph>` API for composing models into a graph structure.
```

File: doc/source/serve/index.md
Comment (edoakes) [inline doc/source/serve/index.md:166]:
```suggestion
    Learn how to :ref:`manage deployments<managing-deployments>`, :ref:`handle dependencies<handling-dependencies>`, :ref:`call deployments via HTTP<serve-http>`, :ref:`call deployments from Python<serve-handle-explainer>`, :ref:`deploy, update, and monitor your application with Ray Serve<serve-deploy-tutorial>`, :ref:`serve multiple ML models<serve-ml-models>`, and :ref:`debug and monitor your Serve applications<serve-monitoring>`.
```

File: doc/source/serve/index.md
Comment (edoakes) [inline doc/source/serve/index.md:178]:
```suggestion
    Follow the tutorials to learn how to integrate Ray Serve with :ref:`Keras and TensorFlow<serve-tensorflow-tutorial>`, :ref:`Scikit-Learn<serve-sklearn-tutorial>`, and :ref:`RLlib<serve-rllib-tutorial>`. Learn how Ray Serve also integrates with :ref:`existing web applications<serve-web-server-integration-tutorial>`
```

File: doc/source/serve/index.md
Comment (edoakes) [inline doc/source/serve/index.md:199]:
```suggestion
    **API Reference**
```

File: doc/source/serve/index.md
Comment (edoakes) [inline doc/source/serve/index.md:141]: suggest to delete this sentence

File: doc/source/serve/index.md
Comment (edoakes) [reply doc/source/serve/index.md:166]: This is very long, I don't think we need to laundry-list every user guide. How about a more generic/concise description like "Learn best practices for common patterns like handling dependencies and calling deployments via HTTP and Python."

**Review by edoakes — COMMENTED** [review]
This looks awesome, thanks so much for the contribution!

**Comment by emjames** [comment]
Thank you for the review  @edoakes ! I've updated the files with your suggested changes.
Let me know what you think !

**Comment by edoakes** [comment]
@emjames looks good, but the linter is failing. You can see the error message here:
https://buildkite.com/ray-project/ray-builders-pr/builds/36682#0181a077-e0b5-4a83-b0a4-4be0440c0b2d/7591-7896

**Comment by emjames** [comment]
I appreciate your feedback. I'm carefully reading the [Getting Involved](https://docs.ray.io/en/latest/ray-contribute/getting-involved.html) to understand how to do this properly.

**Comment by emjames** [comment]
I've been trying to build the docs following the [guide](https://docs.ray.io/en/master/ray-contribute/development.html#building-the-docs) and I'm getting an error:
```
> make html                                                                                                                                              130 ↵
sphinx-build -W -b html -d _build/doctrees   source _build/html
Running Sphinx v4.3.2
Package pickle5 becomes unnecessary in Python 3.8 and above. Its presence may confuse libraries including Ray. Please uninstall the package.
making output directory... done
myst v0.15.2: MdParserConfig(renderer='sphinx', commonmark_only=False, enable_extensions=['dollarmath', 'amsmath', 'deflist', 'html_admonition', 'html_image', 'colon_fence', 'smartquotes', 'replacements'], dmath_allow_labels=True, dmath_allow_space=True, dmath_allow_digits=True, dmath_double_inline=False, update_mathjax=True, mathjax_classes='tex2jax_process|mathjax_process|math|output_area', disable_syntax=[], url_schemes=['http', 'https', 'mailto', 'ftp'], heading_anchors=None, heading_slug_func=None, html_meta=[], footnote_transition=True, substitutions=[], sub_delimiters=['{', '}'], words_per_minute=200)
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 397 source files that are out of date
updating environment: [new config] 397 added, 0 changed, 0 removed
/home/ej/Workspace/EJ/ray/doc/venv/lib/python3.8/site-packages/torch/cuda/__init__.py:83: UserWarning: CUDA initialization: Unexpected error from cudaGetDeviceCount(). Did you run some cuda functions before calling NumCudaDevices() that might have already set an error? Error 804: forward compatibility was attempted on non supported HW (Triggered internally at  ../c10/cuda/CUDAFunctions.cpp:109.)
  return torch._C._cuda_getDeviceCount() > 0
NCCL seems unavailable. Please install Cupy following the guide at: https://docs.cupy.dev/en/stable/install.html.                                              
lz4 not available, disabling sample compression. This will significantly impact RLlib performance. To install lz4, run `pip install lz4`.                      
/home/ej/Workspace/EJ/ray/python/ray/tune/_structure_refactor.py:29: DeprecationWarning: The module `ray.tune.ray_trial_executor` has been moved to `ray.tune.execution.ray_trial_executor` and the old location will be deprecated soon. Please adjust your imports to point to the new location.Example: Do a global search and replace `ray.tune.ray_trial_executor` with `ray.tune.execution.ray_trial_executor`.
  warnings.warn(warning, DeprecationWarning)
/home/ej/Workspace/EJ/ray/python/ray/tune/_structure_refactor.py:29: DeprecationWarning: The module `ray.tune.trial_runner` has been moved to `ray.tune.execution.trial_runner` and the old location will be deprecated soon. Please adjust your imports to point to the new location.Example: Do a global search and replace `ray.tune.trial_runner` with `ray.tune.execution.trial_runner`.
  warnings.warn(warning, DeprecationWarning)
/home/ej/Workspace/EJ/ray/python/ray/tune/_structure_refactor.py:29: DeprecationWarning: The module `ray.tune.trial` has been moved to `ray.tune.experiment.trial` and the old location will be deprecated soon. Please adjust your imports to point to the new location.Example: Do a global search and replace `ray.tune.trial` with `ray.tune.experiment.trial`.
  warnings.warn(warning, DeprecationWarning)
/home/ej/Workspace/EJ/ray/python/ray/tune/_structure_refactor.py:29: DeprecationWarning: The module `ray.tune.suggest` has been moved to `ray.tune.search` and the old location will be deprecated soon. Please adjust your imports to point to the new location.Example: Do a global search and replace `ray.tune.suggest` with `ray.tune.search`.
  warnings.warn(warning, DeprecationWarning)
/home/ej/Workspace/EJ/ray/python/ray/tune/_structure_refactor.py:29: DeprecationWarning: The module `ray.tune.suggest.suggestion` has been moved to `ray.tune.search.searcher` and the old location will be deprecated soon. Please adjust your imports to point to the new location.ATTENTION: This module may have been split or refactored. Please check the contents of `ray.tune.search.searcher` before making changes.
  warnings.warn(warning, DeprecationWarning)
/home/ej/Workspace/EJ/ray/python/ray/tune/_structure_refactor.py:29: DeprecationWarning: The module `ray.tune.sample` has been moved to `ray.tune.search.sample` and the old location will be deprecated soon. Please adjust your imports to point to the new location.Example: Do a global search and replace `ray.tune.sample` with `ray.tune.search.sample`.
  warnings.warn(warning, DeprecationWarning)
/home/ej/Workspace/EJ/ray/python/ray/tune/_structure_refactor.py:29: DeprecationWarning: The module `ray.tune.suggest.bohb` has been moved to `ray.tune.search.bohb` and the old location will be deprecated soon. Please adjust your imports to point to the new location.Example: Do a global search and replace `ray.tune.suggest.bohb` with `ray.tune.search.bohb`.
  warnings.warn(warning, DeprecationWarning)
/home/ej/Workspace/EJ/ray/python/ray/tune/_structure_refactor.py:29: DeprecationWarning: The module `ray.tune.suggest.skopt` has been moved to `ray.tune.search.skopt` and the old location will be deprecated soon. Please adjust your imports to point to the new location.Example: Do a global search and replace `ray.tune.suggest.skopt` with `ray.tune.search.skopt`.
  warnings.warn(warning, DeprecationWarning)
/home/ej/Workspace/EJ/ray/python/ray/tune/_structure_refactor.py:29: DeprecationWarning: The module `ray.tune.suggest.hyperopt` has been moved to `ray.tune.search.hyperopt` and the old location will be deprecated soon. Please adjust your imports to point to the new location.Example: Do a global search and replace `ray.tune.suggest.hyperopt` with `ray.tune.search.hyperopt`.
  warnings.warn(warning, DeprecationWarning)
/home/ej/Workspace/EJ/ray/python/ray/tune/_structure_refactor.py:29: DeprecationWarning: The module `ray.tune.suggest.optuna` has been moved to `ray.tune.search.optuna` and the old location will be deprecated soon. Please adjust your imports to point to the new location.Example: Do a global search and replace `ray.tune.suggest.optuna` with `ray.tune.search.optuna`.
  warnings.warn(warning, DeprecationWarning)
reading sources... [100%] workflows/package-ref                                                                                                                
looking for now-outdated files... none found
pickling environment... done
checking consistency... done
preparing documents... done
writing output... [100%] workflows/package-ref                                                                                                                 
sitemap.xml was generated for URL / in /home/ej/Workspace/EJ/ray/doc/_build/html/sitemap.xml

Warning, treated as error:
/home/ej/Workspace/EJ/ray/doc/venv/lib/python3.8/site-packages/xgboost_ray/sklearn.py:docstring of xgboost_ray.sklearn.RayXGBClassifier:69:unknown document: /tutorials/monotonic
make: *** [Makefile:55: html] Error 2
```

From the `xgboost_ray` [repo](https://github.com/ray-project/xgboost_ray) I found a similar [issue](https://github.com/ray-project/ray/issues/24200) with a [solution](https://github.com/ray-project/xgboost_ray/pull/214) but that solution doesn't solve for `/tutorials/monotonic`. 
What are some links that I should read? How should I proceed?

## Versions
**Python** 3.8.10 & 3.8.13
**OS** Ubuntu 20.04.3 LTS & Fedora 35

## Reproduction
```
git clone https://github.com/ray-project/ray
cd ray/doc
python -m venv venv
source ./venv/bin/activate
pip install -r requirements-doc.txt
make html
```

**Comment by edoakes** [comment]
@emjames we recently updated the guide for building docs from source, can you check if this helps?
https://github.com/ray-project/ray/pull/26045

**Comment by emjames** [comment]
@edoakes Much appreciated!
Indeed, I solved the issue by renaming each occurrence of `key-concepts` in the `/tutorials/` but I want to run the CI locally first so I don't use up resources

**Comment by edoakes** [comment]
@emjames don't worry about using up resources from CI :)

**Comment by emjames** [comment]
@edoakes copy that!
I've updated the docs and all checks are passing

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
@clarkzinzow @jiajiexiao @jjyao @scv119 can we get a stamp for codeowner approval pls?

**Review by scv119 — APPROVED** [review]

**Pull request merged**
